### PR TITLE
add WebClient example

### PR DESCRIPTION
**Pull request by yuluo-yx**
Add WebClient example

### CODE REVIEW

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/RPCType.java [hunk 1]
```diff
@@ -25,6 +25,11 @@ public enum RPCType {
 	/**
 	 * Spring Cloud.
 	 */
-	SpringCloud
+	SpringCloud,
+
+	/**
+	 * Dubbo.
+	 */
+	Dubbo
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/RPCType.java]: Please delete `Dubbo`. We shouldn't include it.

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/WebRequest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.List;
+
+import com.alibaba.cloud.example.common.entity.Product;
+import com.alibaba.cloud.example.common.entity.ResultHolder;
+
+/**
+ * @author yuluo
+ */
+
+public interface WebRequest {
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/WebRequest.java]: The name of `WebRequest` looks like a parameter, called ProductService?

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/strategy/FeignRequestStrategy.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+/**
+ * @author yuluo
+ */
+
+@FeignClient(name = "product")
+public interface FeignRequestStrategy {
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/strategy/FeignRequestStrategy.java]: Called FeignProductService?

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/strategy/RestTemplateRequestStrategy.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * @author yuluo
+ */
+
+@Component
+public class RestTemplateRequestStrategy implements WebRequest {
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/strategy/RestTemplateRequestStrategy.java]: Called RestTemplateProductService?

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/strategy/WebClientRequestStrategy.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+/**
+ * @author yuluo
+ */
+
+@Component
+public class WebClientRequestStrategy implements WebRequest {
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/strategy/WebClientRequestStrategy.java]: Called WebClientProductService?

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/RPCType.java (hunk 1 above)
Comment (yuluo-yx) [reply spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/RPCType.java]: > 请删除`Dubbo`。我们不应该包括它。

If you delete it from the project, the project will not run properly!

File: spring-cloud-alibaba-examples/appactive-example/frontend/src/main/java/com/alibaba/cloud/example/frontend/controller/FrontController.java [hunk 6]
```diff
@@ -138,18 +153,36 @@ public String detailProduct(
...
-						? productDAO.detail(AppContextClient.getRouteId(), id)
-						: productDAO.detailTemplate(AppContextClient.getRouteId(), id));
+
+		if (Objects.equals(hidden, true)) {
+			return productDAO.detailHidden(id);
+		}
+
+		// Determine the method of the request
+		switch (call) {
+		case "feign":
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/frontend/src/main/java/com/alibaba/cloud/example/frontend/controller/FrontController.java]: Capitalize the first letter, and define it in a constants class.

File: spring-cloud-alibaba-examples/appactive-example/frontend/src/main/java/com/alibaba/cloud/example/frontend/controller/FrontController.java [hunk 7]
```diff
@@ -138,18 +153,36 @@ public String detailProduct(
...
+			resultHolder = productDAO.detail(AppContextClient.getRouteId(), id);
+			break;
+		case "rest":
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/frontend/src/main/java/com/alibaba/cloud/example/frontend/controller/FrontController.java]: Capitalize the first letter, and define it in a constants class.

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/strategy/WebClientProductService.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+ */
+
+@Component
+public class WebClientProductService implements ProductService {
+
+	@Autowired(required = false)
+	WebClient webClient;
+
+	@Autowired
+	private FeignProductService feignRequestStrategy;
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/strategy/WebClientProductService.java]: Please modify the variable name to `feignProductService`.

File: spring-cloud-alibaba-examples/appactive-example/frontend/src/main/java/com/alibaba/cloud/example/frontend/controller/FrontController.java [hunk 9]
```diff
@@ -138,18 +153,36 @@ public String detailProduct(
...
+			resultHolder = productDAO.detailTemplate(AppContextClient.getRouteId(), id);
+			break;
+		case "WebClient":
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/frontend/src/main/java/com/alibaba/cloud/example/frontend/controller/FrontController.java]: Define it in a constants class.

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/ProductDAO.java [hunk 10]
```diff
@@ -38,61 +36,48 @@ public class ProductDAO {
 	RestTemplate restTemplate;
 
 	@Autowired
-	private ProductService productService;
+	private FeignProductService feignRequest;
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/ProductDAO.java]: feignProductService.

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/ProductDAO.java [hunk 11]
```diff
@@ -38,61 +36,48 @@ public class ProductDAO {
...
+
+	@Autowired
+	private RestTemplateProductService restTemplateRequest;
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/ProductDAO.java]: restTemplateProductService

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/ProductDAO.java [hunk 12]
```diff
@@ -38,61 +36,48 @@ public class ProductDAO {
...
+
+	@Autowired
+	private WebClientProductService webClientRequest;
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/ProductDAO.java]: webClientProductService

File: spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/ProductDAO.java [hunk 13]
```diff
@@ -38,61 +36,48 @@ public class ProductDAO {
...
-			return restTemplate.getForObject("http://product/detail", ResultHolder.class,
-					params);
-		}
-		return productService.detail(rId, pId);
+		return restTemplateRequest.list();
 	}
 
-	@FeignClient(name = "product")
-	public interface ProductService {
+	public ResultHolder<Product> detailTemplate(String rId, String pId) {
```
Comment (steverao) [inline spring-cloud-alibaba-examples/appactive-example/common/src/main/java/com/alibaba/cloud/example/common/service/ProductDAO.java]: Called `detailByRestTemplate`? Please modify similar problems at the same time, such `detailByWebClient`.

**Review by steverao — APPROVED** [review]
LGTM, thanks @yuluo-yx

**Pull request merged**